src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -954,12 +957,25 @@ impl_install_tree (RpmOstreeTreeComposeContext *self,
       }
   }
 
-  /* Bind metadata from the libdnf context */
-  if (!g_hash_table_contains (self->metadata, "rpmostree.rpmmd-repos"))
-    {
-      g_hash_table_insert (self->metadata, g_strdup ("rpmostree.rpmmd-repos"),
-                           rpmostree_context_get_rpmmd_repo_commit_metadata (self->corectx));
-    }
+
+  /* Optionally bind metadata from the libdnf context */
+  rust::Box<rpmostreecxx::RepoMetadataTarget> repomd_target = (*self->treefile_rs)->get_repo_metadata_target();
+  if (repomd_target->is_inline()) {
+    if (!g_hash_table_contains (self->metadata, "rpmostree.rpmmd-repos"))
```

Comment:
Optional (since it's not a new issue): Would make sense to hoist this to a `#define` or `static const` somewhere.  Perhaps just in this file at least.

Reply:
I feel like if we're going to do that it might be better to do it globally. Maybe I'll try to work that up into a separate PR.